test/integration/src/test/java/io/pravega/test/integration/RestoreBackUpDataRecoveryTest.java

```diff
@@ -175,32 +159,33 @@ protected int getThreadPoolSize() {
         return 100;
     }
 
-    BookKeeperStarter setUpNewBK(int instanceId) throws Exception {
-        return new BookKeeperStarter(instanceId);
-    }
-
     /**
      * Sets up a new BookKeeper & ZooKeeper.
      */
-    private static class BookKeeperStarter implements AutoCloseable {
-        private final int bookieCount = 1;
-        private AtomicReference<BookKeeperConfig> bkConfig = new AtomicReference<>();
-        private AtomicReference<CuratorFramework> zkClient = new AtomicReference<>();
-        private BookKeeperServiceRunner bookKeeperServiceRunner;
-        private AtomicReference<BookKeeperServiceRunner> bkService = new AtomicReference<>();
-        private int bkPort;
-
-        BookKeeperStarter(int instanceId) throws Exception {
+    private static class BookKeeperRunner implements AutoCloseable {
+        private final int bkPort;
+        private final BookKeeperServiceRunner bookKeeperServiceRunner;
+        private final AtomicReference<BookKeeperConfig> bkConfig = new AtomicReference<>();
+        private final AtomicReference<CuratorFramework> zkClient = new AtomicReference<>();
+        private final AtomicReference<BookKeeperServiceRunner> bkService = new AtomicReference<>();
+
+        BookKeeperRunner(int instanceId, int bookieCount) throws Exception {
             bkPort = TestUtils.getAvailableListenPort();
-            val bookiePort = new ArrayList<>(Arrays.asList(TestUtils.getAvailableListenPort()));
-
+            val bookiePorts = new ArrayList<Integer>();
+            for (int i = 0; i < bookieCount; i++) {
+                bookiePorts.add(TestUtils.getAvailableListenPort());
+            }
             this.bookKeeperServiceRunner = BookKeeperServiceRunner.builder()
                     .startZk(true)
                     .zkPort(bkPort)
                     .ledgersPath("/pravega/bookkeeper/ledgers")
-                    .bookiePorts(bookiePort)
+                    .bookiePorts(bookiePorts)
                     .build();
-            this.bookKeeperServiceRunner.startAll();
+            try {
+                this.bookKeeperServiceRunner.startAll();
+            } catch (Exception e) {
+                this.close();
```

Comment:
please rethrow exception otherwise it will just continue as if nothing happened.

Reply:
we need similar try cache block around `this.zkClient.get().start();` as well

Reply:
sure. Throwing it now. Though, `this.zkClient.get().start()` doesn't throw any exception.

Reply:
ok.